src/typechecker/FStar.TypeChecker.TcTerm.fs

```diff
@@ -3838,9 +3867,12 @@ and tc_binder env ({binder_bv=x;binder_qual=imp;binder_attrs=attrs}) =
         | _ -> imp, Env.trivial_guard
     in
     let attrs =
-      attrs |> List.map (fun attr ->
+      attrs |> List.map
+      (fun attr ->
         let attr, _, _ = tc_check_tot_or_gtot_term env attr t_unit "" in
-        attr) in
+        attr)
+    in
+    check_erasable_binder_attributes env attrs t;
```

Comment:
And here too, for binders that do not appear in abstractions (e.g., arguments of inductives, val etc.)